### PR TITLE
3.12 support and no setuptools/wheel on 3.12+

### LINKED ISSUE #2487: Stop installing `setuptools` and `wheel` by default on Python 3.12+ environments
**What's the problem this feature will solve?**

pip can now operate without setuptools in the environment, and it will be removed from the default set of `venv` packages in Python 3.12+ (https://github.com/python/cpython/pull/101039)

**Describe the solution you'd like**

What the title says. :)

**Alternative Solutions**

Not doing this, and being inconsistent with `venv`.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by chrysle**
Note this is my first little code contribution, so please be patient. Closes #2487 

- [x] ran the linter to address style issues (`tox -e fix`)
- [x] wrote descriptive pull request text
- [ ] ensured there are test(s) validating the fix
- [ ] added news fragment in `docs/changelog` folder
- [x] updated/extended the documentation

### CODE REVIEW

File: src/virtualenv/seed/embed/base_embed.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
+import sys
 from abc import ABCMeta
 from pathlib import Path
 
 from ..seeder import Seeder
 from ..wheels import Version
 
 PERIODIC_UPDATE_ON_BY_DEFAULT = True
+PYTHON_VERSION = sys.version_info.minor
```
Comment (gaborbernat) [inline src/virtualenv/seed/embed/base_embed.py:11]: Please inline these globals 👍

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/virtualenv/seed/embed/base_embed.py [hunk 2]
```diff
@@ -84,7 +87,7 @@ def add_parser_arguments(cls, parser, interpreter, app_data):  # noqa: U100
                 dest=f"no_{distribution}",
                 action="store_true",
                 help=f"do not install {distribution}",
-                default=False,
+                default=True if (PYTHON_VERSION >= 12) and (distribution in OPTIONAL_DISTS) else False,
```
[the remark this answers is not in the document]
Comment (gaborbernat) [reply src/virtualenv/seed/embed/base_embed.py:90]: See the interpreter object that's a parameter here

**Comment by chrysle** [comment]
Thanks for the review. Updated.

File: docs/user_guide.rst [hunk 3]
```diff
@@ -118,8 +118,8 @@ at the moment has two types of virtual environments:
 Seeders
 -------
 These will install for you some seed packages (one or more of: :pypi:`pip`, :pypi:`setuptools`, :pypi:`wheel`) that
-enables you to install additional python packages into the created virtual environment (by invoking pip). There are two
-main seed mechanism available:
+enables you to install additional python packages into the created virtual environment (by invoking pip). Installing :pypi:`setuptools` and :pypi:`wheel` is disabled by default on Python 3.12+ environments. There are two
```
Comment (gaborbernat) [inline docs/user_guide.rst:121]: Align at 120 cahracters.

File: src/virtualenv/seed/embed/base_embed.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 from ..seeder import Seeder
 from ..wheels import Version
 
-PERIODIC_UPDATE_ON_BY_DEFAULT = True
+PERIODIC_UPDATE_ON_BY_DEFAULT, OPTIONAL_DISTS = True, {"wheel", "setuptools"}
```
Comment (gaborbernat) [inline src/virtualenv/seed/embed/base_embed.py:9]: Inline OPTIONAL_DISTS; we don't need a global for this

File: src/virtualenv/seed/embed/base_embed.py [hunk 5]
```diff
@@ -84,7 +84,9 @@ def add_parser_arguments(cls, parser, interpreter, app_data):  # noqa: U100
                 dest=f"no_{distribution}",
                 action="store_true",
                 help=f"do not install {distribution}",
-                default=False,
+                default=True
+                if (float(interpreter.version_str.rsplit(".", 1)[0]) >= 12) and (distribution in OPTIONAL_DISTS)
```
Comment (gaborbernat) [inline src/virtualenv/seed/embed/base_embed.py:88]: don't use version_str, use version instead

File: whitelist.txt [hunk 6]
```diff
@@ -54,6 +54,7 @@ deque
 devnull
 distlib
 distro
+dists
```
Comment (gaborbernat) [inline whitelist.txt:57]: remove dists

**Review by gaborbernat — CHANGES_REQUESTED** [review]
You also need a test.

**Comment by chrysle** [comment]
Will look into this. Sorry for producing so much review work.

**Comment by gaborbernat** [comment]
Ah this will get complicated, I'll take it from here 👍

**Comment by chrysle** [comment]
Thank you! I'll try to be more help next time.

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**